include/boost/url/string.hpp

```diff
@@ -55,38 +105,115 @@ class string_value : public string_view
         char*& dest);
 
 public:
+    /** The default allocator type to construct read-only instances of @ref string_value
+
+        An allocator of this class can be instantiated to create
+        multiple read-only string values.
+
+        @note This custom allocator is not an instance of the C++ Allocator named requirement
+
+        @par References
+        @li <a href="https://en.cppreference.com/w/cpp/named_req/Allocator">Allocator (cppreference)</a>
+
+     */
     class allocator;
 
-    inline
+    /** Destructor
+
+        This function decrements the string_value reference counter and
+        destroys the underlying string if no other instance of @ref string_value
+        refers to the same underlying string.
```

Comment:
Upon destruction, ownership of the shared string is released. If there are no more owners, the resources for the string are released.

Reply:
We usually never talk about reference counting, but instead discuss "shared ownership"